### PR TITLE
TASK-895 WS event handler for copy all versions & clone workspace

### PR DESCRIPTION
**Pull request by MrCreosote**
Adds a handler for the copy all versions and clone workspace workspace events.

Error handling still needs some work - currently an error will cause the sub event to be skipped, which is not usually the right thing to do.

### CODE REVIEW

**Comment by MrCreosote** [comment]
Build failure, doesn't seem to be related to the actual code:
```
Patching redis-server init script
git.checkout

1.02s$ git clone --depth=50 https://github.com/kbaseapps/KBaseRelationEngine.git kbaseapps/KBaseRelationEngine

Cloning into 'kbaseapps/KBaseRelationEngine'...

remote: Counting objects: 1300, done.

remote: Compressing objects: 100% (502/502), done.

Receiving objects: 100% (1300/1300), 3.47 MiB | 0 bytes/s, done.

remote: Total 1300 (delta 689), reused 1244 (delta 649), pack-reused 0

Resolving deltas: 100% (689/689), done.

$ cd kbaseapps/KBaseRelationEngine

0.40s$ git fetch origin +refs/pull/4/merge:

remote: Counting objects: 30, done.

remote: Compressing objects: 100% (12/12), done.

remote: Total 30 (delta 16), reused 29 (delta 16), pack-reused 0

Unpacking objects: 100% (30/30), done.

From https://github.com/kbaseapps/KBaseRelationEngine

 * branch            refs/pull/4/merge -> FETCH_HEAD

$ git checkout -qf FETCH_HEAD
rvm

1.02s$ rvm use default

Using /home/travis/.rvm/gems/ruby-2.4.1

$ ruby --version

ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]

$ rvm --version

rvm 1.29.2 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]

$ bundle --version

Bundler version 1.15.1

$ gem --version

2.6.12

0.25s$ rake

rake aborted!

No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
```

**Comment by MrCreosote** [comment]
Workspace side PR: https://github.com/kbase/workspace_deluxe/pull/159

**Comment by MrCreosote** [comment]
@rsutormin I think this is ready for review/discussion.

**Comment by rsutormin** [comment]
@MrCreosote: I think we should switch to WS admin commands in few places:
(1) https://github.com/kbaseapps/KBaseRelationEngine/pull/4/files?w=1#diff-dc17b0a557c9a9e54c7e917aead73f84R107

File: lib/src/kbaserelationengine/events/handler/WorkspaceEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            final List<ObjectSpecification> objs = new LinkedList<>();
+            for (int ver = startVersion; ver < endVersion; ver++) {
+                objs.add(new ObjectSpecification()
+                        .withWsid((long) accessGroupId)
+                        .withObjid(objectID)
+                        .withVer((long) ver));
+            }
+            try {
+                queue.addAll(buildEvents(sourceEvent,
+                        ws.getObjectInfo3(new GetObjectInfo3Params()
```
Comment (rsutormin) [inline lib/src/kbaserelationengine/events/handler/WorkspaceEventHandler.java]: (2) Also here.

File: lib/src/kbaserelationengine/events/handler/WorkspaceEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        try {
+            objid = Long.parseLong(event.getAccessGroupObjectId());
+        } catch (NumberFormatException ne) {
+            //TODO EXP this exception should prevent the event from being processed again
+            throw new IllegalStateException("Illegal workspace object id: " +
+                    event.getAccessGroupObjectId());
+        }
+        try {
+            return buildEvents(event, 
+                    ws.getObjectHistory(new ObjectIdentity()
```
Comment (rsutormin) [inline lib/src/kbaserelationengine/events/handler/WorkspaceEventHandler.java]: (3) And here.

**Comment by rsutormin** [comment]
Actually Gavin thinks it's more convenient to switch to admin commands in separate PR. So I'm merging this one.

**Pull request merged**